src/metabase/driver/mongo/query_processor.clj

```diff
@@ -158,7 +161,26 @@
 
   DateTimeField
   (->rvalue [this]
-    (str \$ (->lvalue this)))
+    ;; omf: date/time fields using native Mongo operators
+    (let [unit (:unit this)
+      field (:field this)]
+      (case (or unit :default)
+        :default         (->rvalue field)
+        :minute          {:$dateToString {:format "%Y-%m-%d %H:%M" :date (->rvalue field)}}
+        :minute-of-hour  {:$minute (->rvalue field)}
+        :hour            {:$dateToString {:format "%Y-%m-%d %H" :date (->rvalue field)}}
+        :hour-of-day     {:$hour (->rvalue field)}
+        :day             {:$dateToString {:format "%Y-%m-%d" :date (->rvalue field)}}
+        :day-of-week     {:$dayOfWeek (->rvalue field)}
+        :day-of-month    {:$dayOfMonth (->rvalue field)}
+        :day-of-year     {:$dayOfYear (->rvalue field)}
+        :week            {:$dateToString {:format "%Y-%m-%d" :date { :$subtract [ (->rvalue field) {:$multiply [ {:$subtract [{:$dayOfWeek (->rvalue field)}, 1]}, 86400000]}] }}}
```

Comment:
in Lisp languages you generally don't put spaces after things like opening brackets